File: callable1/exercises/cmp.cc
#include <cassert>
#include <iostream>

struct student
{
  int m_year;
  std::string m_name;
};

int main()
{
  auto c = [flag = true](const student &a, const student &b)
  {
    return flag ? a.m_name < b.m_name : a.m_year < b.m_year;
  };

  student boe{2, "Boe"}, joe{1, "Joe"};

  struct
  {
    bool flag = true;
    bool operator() (const student &a, const student &b) const
    {
      return flag ? a.m_name < b.m_name : a.m_year < b.m_year;
    }
  } f;

  assert(c(boe, joe));
  assert(f(boe, joe));
}

File: callable1/exercises/cmp.pl
There is given a structure `student` with fields `m_year` of type
`int` and `m_name` of type `string`.

Proszę napisać wyrażenie lambda, które będzie porównywało dwa obiekty
struktury student: `a` i `b`.  Wyrażenie lambda przechwytuje wartość
logiczną, od której zależy wynik porównania: jeżeli przechwyconą
wartością jest:

* `false`, to porównywane są pola `m_year`: `a.m_year < b.m_year`,

* `true`, to porównywane są pola `m_name`: `a.m_name < b.m_name`.

Następnie proszę zaimplementować tą samą funkcjonalność z użyciem
struktury funkcyjnej i funktora.
